examples/vision/sample_size_estimate.py

```diff
@@ -0,0 +1,601 @@
+        if block_name in layer.name:
+            set_trainable = True
+        if set_trainable and not isinstance(layer, layers.BatchNormalization):
+            layer.trainable = True
+            if verbose == 1:
+                print(layer.name, "trainable")
+        else:
+            layer.trainable = False
+            if verbose == 1:
+                print(layer.name, "NOT trainable")
+
+    print("Trainable weights:", len(model.trainable_weights))
+    print("Non-trainable weights:", len(model.non_trainable_weights))
+    return model
+
+
+"""
+# Define augmentation function
+
+We'll use the RandAugment augmentation as shown in a recent Keras code example:
+[RandAugment for Image Classification for Improved
+Robustness](https://https://keras.io/examples/vision/randaugment/).
+"""
+
+rand_aug = iaa.RandAugment(n=3, m=7)
+
+
+def rand_augment(images):
+    """
+    Randomly augments a batch of image tensors using imgaug library.
+
+    Arguments
+    ---------
+    images:       TensorFlow tensors to augment, must be batched
+
+    """
+    # Convert back to uint8 and convert tensor to numpy values
+    images = tf.cast(images, tf.uint8)
+    images = rand_aug(images=images.numpy())
+
+    return images
+
+
+"""
+# Define iterative subsampling functions
+
+To iteratively train a model over several subsample sets we need to create some functions
+for splitting the dataset and training the model.
+
+"""
+
+
+def create_subsets(dataset, percentage):
+    """
+    Creates a subset from a dataset based on a specified percentage. This subset
+    is then divided into training (90%) and validation (10%) sets. Output contains
+    the training set, validation set and number of subsamples.
+
+    Arguments
+    ---------
+    dataset:      Dataset in tf.data format.
+    percentage:   Number between 0 and 100, excluding limits (0 > x > 100).
+
+    """
+
+    ds_subset = dataset.take(int(nr_train_samples * percentage))
+
+    # Calculate subset length
+    nr_of_subsamples = ds_subset.cardinality().numpy()
+    print("Subset length:", nr_of_subsamples)
+
+    # Split off 90% for training and 10% for validation
+    train_val_split = int(nr_of_subsamples * 0.9)
+    train_ds = ds_subset.take(train_val_split)
+    print("Nr of training samples:", train_ds.cardinality().numpy())
+
+    val_ds = ds_subset.skip(train_val_split)
+    print("Nr of validation samples:", val_ds.cardinality().numpy())
+
+    # Prepare the training set for training, apply random augmentations
+    train_ds = (
+        train_ds.map(
+            lambda img, lab: (tf.image.resize(img, image_size), lab),
+            num_parallel_calls=AUTO,
+        )
+        .map(
+            lambda img, lab: (img, tf.one_hot(lab, num_classes)),
+            num_parallel_calls=AUTO,
+        )
+        .batch(batch_size)  # First batch images before augmenting
+        .map(
+            lambda img, lab: (
+                tf.py_function(rand_augment, [img], [tf.float32])[0],
```

Comment:
With image augmentation layers, you won't need a py_function and your pipeline will be more performant.